Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -25,23 +25,112 @@ static void Main(string[] args)
                 new[] { 93, 96, 13, 95, 98, 18, 59, 49, 86 }, 
                 new[] { "f", "Pt", "PT", "fT", "Cp", "C", "t", "", "cCp", "ttp", "PCFt", "P", "pCt", "cP", "Pc" }, 
                 new[] { 2, 6, 6, 2, 4, 4, 5, 0, 5, 5, 6, 6, 3, 5, 6 });
-            Console.ReadKey(true);
+            
         }
 
-        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
-        {
+        private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int n = protein.Length;
+            var calories = new int[n];
+
+            for(int i = 0;i <n; i++)

```

Comment:
try to put some structure in your code. In this case, you can create a class for Meal having integer properties for Protein, Carb etc. Convert the input into a List<Meal> so that it is easy to operate upon. Then you can use LINQ operations like OrderBy() Thenby() First() Select() for getting required sorting and selection.